File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> li=new ArrayList<List<Integer>>();
        List<TreeNode> lii;
        List<List<TreeNode>>queue=new ArrayList<List<TreeNode>>();
        if(root!=null){
            lii=new ArrayList<TreeNode>();
            lii.add(root);
            queue.add(lii);
            while(true){
                Iterator<TreeNode> itr=queue.get(queue.size()-1).listIterator();
                //System.out.print(queue.get(queue.size()-1));
                lii=new ArrayList<TreeNode>();
                while(itr.hasNext()){
                    TreeNode u=itr.next();
                    TreeNode v_l=u.left;
                    TreeNode v_r=u.right;
                    if(v_l!=null) lii.add(v_l);
                    if(v_r!=null) lii.add(v_r);
                    
                }
                if(!lii.isEmpty()){
                    queue.add(lii);
                }
                else break;
            }
            List<Integer> numli;
            Iterator<List<TreeNode>> itr=queue.listIterator();
            while(itr.hasNext()){
                numli=new ArrayList<Integer>();
                Iterator<TreeNode> it=itr.next().listIterator();
                while(it.hasNext()){
                    numli.add(it.next().val);
                }
                li.add(numli);
            }
        }
        return li;
        
    }
}

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<pair<TreeNode*,int>> q;
        vector<vector<int>> v;
        pair<TreeNode*,int> f;
        if(root!=NULL){
             q.push({root,0});
            while(not q.empty()){
                f=q.front();
                q.pop();
                if(v.size()-1!=f.second){
                    vector<int> v1;
                    v1.push_back(f.first->val);
                    v.push_back(v1);
                }
                else{
                    v[v.size()-1].push_back(f.first->val);
                }
                if(f.first->left)
                    q.push({f.first->left,f.second+1});
                if(f.first->right)
                    q.push({f.first->right,f.second+1});
            }
        }
        return v;
    
    }
};